subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/internal/precompiled/PrecompiledGroovyPluginsPlugin.java

```diff
@@ -89,9 +98,20 @@ private void exposeScriptsAsPlugins(Project project) {
         pluginSourceSet.getOutput().dir(extractPluginRequests.flatMap(ExtractPluginRequestsTask::getExtractedPluginRequestsClassesStagingDirectory));
     }
 
-    private void declarePluginMetadata(GradlePluginDevelopmentExtension pluginExtension, List<PrecompiledGroovyScript> scriptPlugins) {
+    private void validateScriptPlugin(Project project, PrecompiledGroovyScript scriptPlugin) {
+        if (scriptPlugin.getId().equals(CORE_PLUGIN_NAMESPACE) || scriptPlugin.getId().startsWith(CORE_PLUGIN_PREFIX)) {
+            throw new GradleException(String.format("Precompiled plugin can't start with '%s': '%s'.\n\n%s",
```

Comment:
Please make this a similar kind of error as above.

Reply:
done